Formula/snapcraft.rb

```diff
@@ -164,8 +235,14 @@ class Snapcraft < Formula
   end
 
   resource "PyNaCl" do
-    url "https://files.pythonhosted.org/packages/cf/5a/25aeb636baeceab15c8e57e66b8aa930c011ec1c035f284170cacb05025e/PyNaCl-1.4.0.tar.gz"
-    sha256 "54e9a2c849c742006516ad56a88f5c74bf2ce92c9f67435187c3c5953b346505"
+    on_macos do
+      url "https://files.pythonhosted.org/packages/cf/5a/25aeb636baeceab15c8e57e66b8aa930c011ec1c035f284170cacb05025e/PyNaCl-1.4.0.tar.gz"
+      sha256 "54e9a2c849c742006516ad56a88f5c74bf2ce92c9f67435187c3c5953b346505"
+    end
+    on_linux do
+      url "https://files.pythonhosted.org/packages/61/ab/2ac6dea8489fa713e2b4c6c5b549cc962dd4a842b5998d9e80cf8440b7cd/PyNaCl-1.3.0.tar.gz"
+      sha256 "0c6100edd16fefd1557da078c7a31e7b7d7a52ce39fdca2bec29d4f7b6e7600c"
+    end
```

Comment:
Apparently creating Python resources like this breaks `brew update-python-resources`.  See the discussion in https://github.com/Homebrew/homebrew-core/pull/82725.  So we'll have to grab this resource inside `on_macos` and `on_linux` blocks instead.

Reply:
`brew update-python-resources` doesn't work here since main source URL is GitHub not PyPI. We only support auto-update logic via PyPI.

PyPI only has older v4.8.1: https://pypi.org/project/snapcraft/

Reply:
I see. It sounds to me like we will need to make some eventual improvments to support OS-specific and non-PyPI sources, but that should be the focus of a dedicated issue for `brew`.

Reply:
On separate note, I wonder if we should try PyNaCl v1.4.0 on Linux.
Upstream uses v1.3.0 due to https://github.com/snapcore/snapcraft/pull/3483
It is possible that Homebrew `libsodium` will work compared to the particular system `libsodium` that issue was raised on.

Reply:
Giving it a try now. If test passes, then we should be fine as upstream issue failed on `snapcraft --help`